ethexe/rpc/src/apis/injected.rs

```diff
@@ -121,65 +166,29 @@ impl InjectedServer for InjectedApi {
         })
     }
 
-    async fn send_transaction_and_watch(
+    fn spawn_promise_subscriber(
         &self,
-        pending: PendingSubscriptionSink,
-        transaction: RpcOrNetworkInjectedTx,
-    ) -> SubscriptionResult {
-        let tx_hash = transaction.tx.data().to_hash();
-        tracing::trace!(%tx_hash, ?transaction, "Called injected_sendTransactionAndWatch");
-
-        // Checks, that transaction wasn't already send.
-        if self.promise_waiters.get(&tx_hash).is_some() {
-            tracing::warn!(tx_hash = ?tx_hash, "transaction was already sent");
-            return Err(
-                format!("transaction with the same hash was already sent: {tx_hash}").into(),
-            );
-        }
-
-        let (response_sender, response_receiver) = oneshot::channel();
-        let (promise_sender, promise_receiver) = oneshot::channel();
-
-        let event = RpcEvent::InjectedTransaction {
-            transaction,
-            response_sender,
-        };
-
-        if let Err(err) = self.rpc_sender.send(event) {
-            tracing::error!(
-                "Failed to send `RpcEvent::InjectedTransaction` event task: {err}. \
-                The receiving end in the main service might have been dropped."
-            );
-            return Err(errors::internal().into());
-        }
-
-        if let InjectedTransactionAcceptance::Reject { reason } = response_receiver.await? {
-            tracing::trace!(
-                tx_hash = ?tx_hash,
-                reject_reason = ?reason,
-                "reject injected transaction"
-            );
-            pending.reject(errors::bad_request(reason)).await;
-            return Ok(());
-        }
-
-        let subscription_sink = match pending.accept().await {
-            Ok(sink) => sink,
-            Err(err) => {
-                tracing::warn!(
-                    "failed to accept subscription for injected transaction promise: {err}"
-                );
-                return Ok(());
-            }
-        };
-
-        tracing::trace!(?tx_hash, "Accept transition, start promise waiter");
-
-        self.promise_waiters.insert(tx_hash, promise_sender);
-
+        sink: SubscriptionSink,
+        receiver: oneshot::Receiver<SignedPromise>,
+        tx_hash: HashOf<InjectedTransaction>,
+    ) {
+        let promise_waiters = self.promise_waiters.clone();
         tokio::spawn(async move {
```

Comment:
For future, not for now. Looks like tokio::spawn is not necessary here, but removing it needs some refactoring

Reply:
sending second message - can be done in RPC service polling